org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java

```diff
@@ -418,30 +442,1025 @@ public static String mergeIntoCommaSeparatedList(String currentString, Set<Strin
+		 * Attributes to this Map. Duplicate mappings will be replaced.
+		 *
+		 * @param attr the Attributes to be stored in this map
+		 * @exception ClassCastException if attr is not an Attributes
+		 */
+		public void putAll(Map<?, ?> attr) {
+			// ## javac bug?
+			if (!Attributes.class.isInstance(attr))
+				throw new ClassCastException();
+			for (Map.Entry<?, ?> me : (attr).entrySet())
+				put(me.getKey(), me.getValue());
+		}
+
+		/**
+		 * Removes all attributes from this Map.
+		 */
+		public void clear() {
+			map.clear();
+		}
+
+		/**
+		 * Returns the number of attributes in this Map.
+		 */
+		public int size() {
+			return map.size();
+		}
+
+		/**
+		 * Returns true if this Map contains no attributes.
+		 */
+		public boolean isEmpty() {
+			return map.isEmpty();
+		}
+
+		/**
+		 * Returns a Set view of the attribute names (keys) contained in this
+		 * Map.
+		 */
+		public Set<Object> keySet() {
+			return map.keySet();
+		}
+
+		/**
+		 * Returns a Collection view of the attribute values contained in this
+		 * Map.
+		 */
+		public Collection<Object> values() {
+			return map.values();
+		}
+
+		/**
+		 * Returns a Collection view of the attribute name-value mappings
+		 * contained in this Map.
+		 */
+		public Set<Map.Entry<Object, Object>> entrySet() {
+			return map.entrySet();
+		}
+
+		/**
+		 * Compares the specified Attributes object with this Map for equality.
+		 * Returns true if the given object is also an instance of Attributes
+		 * and the two Attributes objects represent the same mappings.
+		 *
+		 * @param o the Object to be compared
+		 * @return true if the specified Object is equal to this Map
+		 */
+		@Override
+		public boolean equals(Object o) {
+			return map.equals(o);
+		}
+
+		/**
+		 * Returns the hash code value for this Map.
+		 */
+		@Override
+		public int hashCode() {
+			return map.hashCode();
+		}
+
+		/**
+		 * Returns a copy of the Attributes, implemented as follows:
+		 * 
+		 * <pre>
+		 * public Object clone() {
+		 * 	return new Attributes(this);
+		 * }
+		 * </pre>
+		 * 
+		 * Since the attribute names and values are themselves immutable, the
+		 * Attributes returned can be safely modified without affecting the
+		 * original.
+		 */
+		@Override
+		public Object clone() {
+			return new Attributes(this);
+		}
+
+		/*
+		 * Writes the current attributes to the specified data output stream.
+		 * XXX Need to handle UTF8 values and break up lines longer than 72
+		 * bytes
+		 */
+		@SuppressWarnings("deprecation")
+		void write(DataOutputStream os) throws IOException {
+			for (Entry<Object, Object> e : entrySet()) {
+				StringBuffer buffer = new StringBuffer(((Name) e.getKey()).toString());
+				buffer.append(": ");
+
+				String value = (String) e.getValue();
+				if (value != null) {
+					byte[] vb = value.getBytes("UTF8");
+					value = new String(vb, 0, 0, vb.length);
+				}
+				buffer.append(value);
+
+				buffer.append("\r\n");
+				MergeableManifest.make72Safe(buffer);
```

Comment:
Other places call make512Safe - should we adjust this location, too?

Reply:
I have tried to modify existing code as less as possible. Checked again and found that it is dead code. Will delete the dead code.